scipy/stats/tests/test_distributions.py

```diff
@@ -3852,6 +3852,87 @@ def test_cdf_ppf_sf_isf_tail(self, x, a, expected):
         isf = stats.dgamma.isf(expected, a)
         assert_allclose(isf, -x, rtol=5e-15)
 
+    @pytest.mark.parametrize("a, ref",
+        [(1.5, 2.0541199559354118267922018421759071394202272087878735),
+        (1.3, 1.9357296377121247644586320317632592032856887302320274),
+        (1.1, 1.7856502333412133408733521579693880954302277243835155)])
+    def test_entropy(self, a, ref):
+        #The reference values were calculated with mpmath:
+        #def entropy_dgamma(a):
+        #   def pdf(x):
+        #       A = 1 / (2 * mp.gamma(a))
+        #       B = mp.fabs(x) ** (a - 1)
+        #       C = mp.exp(-mp.fabs(x))
+        #       h = A * B * C
+        #       return h
+        #
+        #   return -mp.quad(lambda t: pdf(t) * mp.log(pdf(t)), [-mp.inf, mp.inf])
+        assert_allclose(stats.dgamma.entropy(a), ref, rtol=1e-14)
+
+    @pytest.mark.parametrize("a, ref",
+    [(1e-100, -1.0e+100),
+    (1e-95, -1.0e+95),
+    (1e-90, -1.0e+90),
+    (1e-85, -1.0e+85),
+    (1e-80, -1.0e+80),
+    (1e-75, -1.0e+75),
+    (1e-70, -1.0e+70),
+    (1e-65, -1.0e+65),
+    (1e-60, -1.0e+60),
+    (1e-55, -1.0e+55),
+    (1e-50, -1.0e+50),
+    (1e-45, -1.0e+45),
+    (1e-40,-1.0e+40),
+    (1e-35, -1.0e+35),
+    (1e-30, -1.0e+30),
+    (1e-25, -1.0e+25),
+    (1e-20, -1.0e+20),
+    (1e-15, -999999999999964.0),
+    (1e-10, -9999999975.85822),
+    (1e-5, -99987.3711165702),
+    (10000000000.0, 13.625),
+    (1000000000000000.0, 19.38147391122),
+    (1e+20, 25.1379366437051),
+    (1e+25, 30.8943993761902),
+    (1e+30, 36.6508621086753),
+    (1e+35, 42.4073248411604),
+    (1e+40, 48.1637875736455),
+    (1e+45, 53.9202503061306),
+    (1e+50, 59.6767130386158),
+    (1e+55, 65.4331757711009),
+    (1e+60, 71.189638503586),
+    (1e+65, 76.9461012360711),
+    (1e+70, 82.7025639685562),
+    (1e+75, 88.4590267010413),
+    (1e+80, 94.2154894335264),
+    (1e+85, 99.9719521660116),
+    (1e+90, 105.728414898497),
+    (1e+95, 111.484877630982),
+    (1e+100, 117.241340363467)])
```

Comment:
```suggestion
    [(1e-10, -9999999975.858217),
     (1e-5, -99987.3711165702),
     (1e1, 3.2292013590409248),
     (1e4, 6.717222565586032),
     (1e10, 13.625011178701513),
```

Reply:
Feel free to add up to two more test cases but having more than 30 tests for such a small function  is just way too much. I do not think any other function of the univariate distributions is tested that extensively.

Reply:
Ok, I removed most of the tests and added 1e4, but we also need to keep the tests for 1e100 and 1e-100.